nemo/collections/speechlm/models/speech_to_text_llm_model.py

```diff
@@ -0,0 +1,1127 @@
+        audio_signal_length: Optional[torch.Tensor] = None,
+        processed_signal: Optional[torch.Tensor] = None,
+        processed_signal_length: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+        num_audios: Optional[torch.Tensor] = None,
+        context_start_idx: Optional[List[List[int]]] = None,
+        inference_params: Optional[InferenceParams] = None,
+    ) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+        output = self.module(
+            input_ids=input_ids,
+            input_length=input_length,
+            loss_mask=loss_mask,
+            attention_mask=attention_mask,
+            audio_signal=audio_signal,
+            audio_signal_length=audio_signal_length,
+            processed_signal=processed_signal,
+            processed_signal_length=processed_signal_length,
+            labels=labels,
+            num_audios=num_audios,
+            context_start_idx=context_start_idx,
+            inference_params=inference_params,
+        )
+        return output
+
+    def freeze_llm(self):
+        module = self.module
+        while not hasattr(module, "language_model"):
+            module = module.module
+        self.freeze_module(module.language_model)
+
+    def freeze_speech(self):
+        module = self.module
+        while not hasattr(module, "speech_model"):
+            module = module.module
+        self.freeze_module(module.speech_model)
+
+    def freeze_modality_adapter(self):
+        module = self.module
+        while not hasattr(module, "modality_adapter"):
+            module = module.module
+        self.freeze_module(module.modality_adapter)
+
+    def unfreeze_llm(self):
+        module = self.module
+        while not hasattr(module, "language_model"):
+            module = module.module
+        self.unfreeze_module(module.language_model)
+
+    def unfreeze_speech(self):
+        module = self.module
+        while not hasattr(module, "speech_model"):
+            module = module.module
+        self.unfreeze_module(module.speech_model)
+
+    def unfreeze_modality_adapter(self):
+        module = self.module
+        while not hasattr(module, "modality_adapter"):
+            module = module.module
+        self.unfreeze_module(module.modality_adapter)
+
+    def data_step(self, dataloader_iter) -> Dict[str, torch.Tensor]:
+        return self.config.data_step_fn(dataloader_iter)
+
+    def forward_step(self, batch) -> torch.Tensor:
+        return self.config.forward_step_fn(self, batch)
+
+    def training_step(self, batch, batch_idx=None) -> torch.Tensor:
+        # In mcore the loss-function is part of the forward-pass (when labels are provided)
+        return self.forward_step(batch)
+
+    def validation_step(self, batch, batch_idx=None) -> torch.Tensor:
+        return self.inference_step(batch, mode='validation')
+
+    @property
+    def _metrics_require_string2category_map(self):
```

Comment:
f1 and accuracy for speech metrics?

Reply:
those were kept following the GPT SFT model in nlp collection, can be used for multiple choice QA